RecoHGCal/TICL/plugins/TrackstersMergeProducer.cc

```diff
@@ -511,8 +512,8 @@ void TrackstersMergeProducer::produce(edm::Event &evt, const edm::EventSetup &es
       tmpCandidate.setPdgId(211 * track.charge());
       float energy = std::sqrt(track.pt() * track.pt() + mpion2);
```

Comment:
should this actually use track.p ?
```suggestion
      float energy = std::sqrt(track.p() * track.p() + mpion2);
```
IIUC, this way even the original p4 definition would be correct.
or is there some good reason to keep raw energy to be this "transverse" energy?